utils.h

```diff
@@ -52,19 +52,24 @@ std::string gpt_random_prompt(std::mt19937 & rng);
 // Vocab utils
 //
 
+struct token_score {
```

Comment:
this is confusingly named, same with token_t. the type is only used inside gpt_vocab, so why not nest it.

Reply:
also gpt_vocab is token_t already in this case

Reply:
Thanks!

What name can I give this struct?
![image](https://user-images.githubusercontent.com/9311162/226493299-0bd9c340-1143-4a64-8a02-401d893d7433.png)

Reply:
hm, first thought was token_t, but that is too close to token, so, just leave it as token_score.

Reply:
This compiler version seems to not accept token token;

```
/home/runner/work/llama.cpp/llama.cpp/utils.h:68:15: error: declaration of ‘gpt_vocab::token gpt_vocab::token_score::token’ changes meaning of ‘token’ [-fpermissive]
   68 |         token token;
      |               ^~~~~
/home/runner/work/llama.cpp/llama.cpp/utils.h:65:11: note: ‘token’ declared here as ‘using token = std::string’
   65 |     using token = std::string;
```

Should I rename the using token = std::string; to token_t?

Reply:
The quickest and simplest fix to that would be to just rename the data member to `tok`.